tests/test_access_route.py

```diff
@@ -0,0 +1,74 @@
+from falcon.request import Request
+import falcon.testing as testing
+
+
+class TestAccessRoute(testing.TestBase):
+
+    def test_remote_addr_only(self):
+        req = Request(testing.create_environ(
+            host='example.com',
+            path='/access_route',
+            headers={
+                'Forwarded': ('for=192.0.2.43, for="[2001:db8:cafe::17]:555",'
+                              'for="unknown", by=_hidden,for="\\"\\\\",'
+                              'for="198\\.51\\.100\\.17\\:1236";'
+                              'proto=https;host=example.com')
+            }))
+        self.assertEqual(req.remote_addr, '127.0.0.1')
+
+    def test_rfc_forwarded(self):
+        req = Request(testing.create_environ(
+            host='example.com',
+            path='/access_route',
+            headers={
+                'Forwarded': ('for=192.0.2.43,for=,'
+                              'for="[2001:db8:cafe::17]:555",'
+                              'for="unknown", by=_hidden,for="\\"\\\\",'
```

Comment:
Would you mind adding an additional test that includes a `for` value with a mixture of regular chars, escaped backslashes, and  escaped ASCII chars? It might make sense to do this in `tests/test_utils.py` directly against `uri.unquote_string()`